frontend/webapp/hooks/destinations/useDestinationFormData.ts

```diff
@@ -1,137 +1,147 @@
-import { useState, useEffect, useMemo, useCallback, useRef } from 'react';
-import { safeJsonParse } from '@/utils';
-import { useDrawerStore } from '@/store';
+import { useState, useEffect } from 'react';
+import { DrawerBaseItem } from '@/store';
 import { useQuery } from '@apollo/client';
-import { useConnectDestinationForm } from '@/hooks';
 import { GET_DESTINATION_TYPE_DETAILS } from '@/graphql';
-import { DynamicField, ActualDestination, isActualDestination, DestinationDetailsResponse, SupportedDestinationSignals, DestinationDetailsField } from '@/types';
-
-const DEFAULT_SUPPORTED_SIGNALS: SupportedDestinationSignals = {
-  logs: { supported: false },
-  metrics: { supported: false },
-  traces: { supported: false },
-};
-
-export function useDestinationFormData() {
-  const [dynamicFields, setDynamicFields] = useState<DynamicField[]>([]);
-  const [supportedSignals, setSupportedSignals] = useState<SupportedDestinationSignals>(DEFAULT_SUPPORTED_SIGNALS);
-  const [exportedSignals, setExportedSignals] = useState({
+import { useConnectDestinationForm, useNotify } from '@/hooks';
+import { ACTION, FORM_ALERTS, NOTIFICATION, safeJsonParse } from '@/utils';
+import type { DynamicField, DestinationDetailsResponse, DestinationInput, DestinationTypeItem, ActualDestination, SupportedDestinationSignals } from '@/types';
+
+const INITIAL: DestinationInput = {
+  type: '',
+  name: '',
+  exportedSignals: {
     logs: false,
     metrics: false,
     traces: false,
-  });
+  },
+  fields: [],
+};
 
-  const destination = useDrawerStore(({ selectedItem }) => selectedItem);
-  const shouldSkip = !isActualDestination(destination?.item);
-  const destinationType = isActualDestination(destination?.item) ? destination.item.destinationType.type : null;
+export function useDestinationFormData(params?: { destinationType?: string; supportedSignals?: SupportedDestinationSignals; preLoadedFields?: string | DestinationTypeItem['fields'] }) {
+  const { destinationType, supportedSignals, preLoadedFields } = params || {};
+  const notify = useNotify();
 
-  const { buildFormDynamicFields } = useConnectDestinationForm();
+  const [formData, setFormData] = useState({ ...INITIAL });
+  const [dynamicFields, setDynamicFields] = useState<DynamicField[]>([]);
 
-  const { data: destinationFields } = useQuery<DestinationDetailsResponse>(GET_DESTINATION_TYPE_DETAILS, {
-    variables: { type: destinationType },
-    skip: shouldSkip,
-  });
+  const handleFormChange = (key: keyof typeof INITIAL | string, val: any) => {
+    const [parentKey, childKey] = key.split('.');
```

Comment:
Why does this need to be split by "."?

Reply:
in the case that "exportedSignals" have been changed, it's an object so they children are targeted as: "exportedSignals.logs"